dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/dataitems/DataItemQueryTests.java

```diff
@@ -0,0 +1,297 @@
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hisp.dhis.ApiTest;
+import org.hisp.dhis.actions.LoginActions;
+import org.hisp.dhis.actions.dataitem.DataItemActions;
+import org.hisp.dhis.dto.ApiResponse;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test cases related to GET "dataItems" endpoint. The tests and assertions are
+ * based on the file "setup/metadata.json" => "programIndicators",
+ * "dataElements".
+ * 
+ * The test cases using default pagination will imply "paging=true", which is
+ * the default when "paging" is omitted.
+ *
+ * @author maikel arabori
+ */
+public class DataItemQueryTests extends ApiTest
+{
+    private static final int FOUND = 302;
+
+    private static final int NOT_FOUND = 404;
+
+    private static final int CONFLICT = 409;
+
+    private DataItemActions dataItemActions;
+
+    @BeforeAll
+    public void before()
+    {
+        dataItemActions = new DataItemActions();
+
+        login();
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPagination()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get();
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
```

Comment:
for the sake of reporting, I would use rest-assured assertions here. For example: 
`response.validate().body("nameableObjects.dimensionItemType", hasSize(greaterOrEqualTo(1)))`. If you do so, upon failure the response will be printed out and the assertion message will look more meaningful than only "expected true, but found false"

Reply:
Done